Task05/src/main.cpp

```diff
@@ -0,0 +1,227 @@
+#include <iostream>
+#include <fstream>
+#include <unordered_map>
+#include <string>
+#include <sstream>
+#include <cmath>
+#include <functional>
+#include <algorithm>
+#include <vector>
+#include <array>
+#include "frequencies.h"
+
+
+
```

Comment:
your code is very complex!!, consider using classes and make the code more clean and short